lms/views/api/canvas/authorize.py

```diff
@@ -73,13 +81,18 @@ def oauth2_redirect(self):
         route_name="canvas_api.authorize",
     )
     def oauth2_redirect_error(self):
-        self.request.response.status_code = 500
+        template_variables = {
+            "invalid_scope": self.request.params.get("error") == "invalid_scope",
+            "details": self.request.params.get("error_description"),
+            "scopes": self.scopes,
+        }
 
-        authorization_param = (
-            BearerTokenSchema(self.request).authorization_param(self.request.lti_user),
-        )
-        return {
-            "authorize_url": self.request.route_url(
-                "canvas_api.authorize", _query=[("authorization", authorization_param)]
+        if self.request.lti_user:
+            authorization_param = BearerTokenSchema(self.request).authorization_param(
+                self.request.lti_user
             )
-        }
+            template_variables["authorize_url"] = self.request.route_url(
+                "canvas_api.authorize", _query=[("authorization", authorization_param)],
+            )
```

Comment:
If we're still logged in (`request.lti_user` is still set) then sent the `authorize_url` so that we remain logged in. This makes the <kbd>Try again</kbd> button keep working. But sometimes we've already lost `request.lti_user` by this point and can't do this